pkg/virt-handler/heartbeat/ksm.go

```diff
@@ -249,32 +249,33 @@ func getIntParam(node *v1.Node, param string, defaultValue, lowerBound, upperBou
 // will set the outcome value to the n.KSM struct
 // If the node labels match the selector terms, the ksm will be enabled.
 // Empty Selector will enable ksm for every node
-func handleKSM(node *v1.Node, clusterConfig *virtconfig.ClusterConfig) (bool, bool) {
-	available, running := loadKSM()
+func handleKSM(node *v1.Node, clusterConfig *virtconfig.ClusterConfig) (ksmHandlerEnabled, ksmEnabledByUs bool) {
+	var available bool
+	available, ksmHandlerEnabled = loadKSM()
 	if !available {
-		return running, false
+		return ksmHandlerEnabled, false
 	}
 
 	ksmConfig := clusterConfig.GetKSMConfiguration()
 	if ksmConfig == nil {
-		if disableKSM(node, running) {
+		if disableKSM(node, ksmHandlerEnabled) {
 			return false, false
 		} else {
-			return running, false
+			return ksmHandlerEnabled, false
 		}
 	}
```

Comment:
I think that this code would be more clear:
```
	if ksmConfig == nil && ksmHandlerEnabled {
		disableKSM(node, ksmHandlerEnabled)
		return false, false
	} else if ksmConfig == nil {
		return false, false
	}
```
And remove this check from disableKSM.
or even better:
```
if ksmConfig == nil{
  disableKSM(node, ksmHandlerEnabled)
  return false, false
} 
```

And remove this check from `disableKSM` i think that using os.WriteFile in Go with the same data will not result in an error.

Reply:
Thank you! Much better